{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "90742ef1-eb73-460c-a562-de753dadc038",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pathlib\n",
    "import sys\n",
    "sys.path.append('lib')  \n",
    "import Building_Literature_Embedding_Model as edg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "75384484-4caa-4670-ab24-e802e711546b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define hyper-parameters\n",
    "context_size = 4\n",
    "min_count = 3\n",
    "min_size = 2\n",
    "dimension = 128\n",
    "num_sampled = 64\n",
    "batch_size = 256 \n",
    "epoch = 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "43fcffe5-60a9-4315-bbfc-2d45c0852299",
   "metadata": {},
   "outputs": [],
   "source": [
    "root_path = '../data/preprocessed_data'\n",
    "output='../data/tutorial_literature_embedding_model'  \n",
    "  \n",
    "vocab_dir = output + '/vocab/'\n",
    "preprocessed_path = root_path + '/preprocessed/data.doc.txt'\n",
    "model_path = output \n",
    "logs_dir = vocab_dir+'/logs'\n",
    "gene2doc_dir = logs_dir+'/gene2doc' \n",
    "\n",
    "pathlib.Path(logs_dir).mkdir(parents=True, exist_ok=True)\n",
    "pathlib.Path(gene2doc_dir).mkdir(parents=True, exist_ok=True) \n",
    "pathlib.Path(model_path).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cfbc1543-a6e0-42d5-9f92-c22d8605c385",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==== Generating Training Data for Literature Embedding Model\n",
      "==== Creating Vocabulary ===\n",
      "241315\n"
     ]
    }
   ],
   "source": [
    "print(\"==== Generating Training Data for Literature Embedding Model\")\n",
    "eg=edg.building_embedding_model()\n",
    "eg.setting(preprocessed_path, vocab_dir, logs_dir, gene2doc_dir)\n",
    "\n",
    "print(\"==== Creating Vocabulary ===\")\n",
    "eg.creating_vocab(min_count = min_count, min_size = min_size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a9009adb-c501-44bd-810a-ad25f26f08b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Checking If Data Generation Is Correct ===\n",
      "== Examples:  #LINC01128\n",
      "integrated -> #LINC01128\n",
      "analysis -> #LINC01128\n",
      "risk -> #LINC01128\n",
      "score -> #LINC01128\n",
      "analysis -> #LINC01128\n",
      "risk -> #LINC01128\n",
      "score -> #LINC01128\n",
      "predicting -> #LINC01128\n",
      "risk -> #LINC01128\n",
      "score -> #LINC01128\n",
      "predicting -> #LINC01128\n",
      "prognosis -> #LINC01128\n",
      "score -> #LINC01128\n",
      "predicting -> #LINC01128\n",
      "prognosis -> #LINC01128\n",
      "cerna -> #LINC01128\n",
      "predicting -> #LINC01128\n",
      "prognosis -> #LINC01128\n",
      "cerna -> #LINC01128\n",
      "network -> #LINC01128\n",
      "prognosis -> #LINC01128\n",
      "cerna -> #LINC01128\n",
      "network -> #LINC01128\n",
      "differentially -> #LINC01128\n",
      "cerna -> #LINC01128\n",
      "network -> #LINC01128\n",
      "differentially -> #LINC01128\n",
      "expressed -> #LINC01128\n",
      "network -> #LINC01128\n",
      "differentially -> #LINC01128\n"
     ]
    }
   ],
   "source": [
    "print(\"=== Checking If Data Generation Is Correct ===\")\n",
    "eg.checking_gene2doc_generation(context_size) # Fig. 2 in our visualization paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8a8e2543-f76d-4198-b7af-590cd8a8e0db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cbow  19526\n",
      "starting making data\n",
      "0 gene2doc\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:184: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:189: The name tf.truncated_normal is deprecated. Please use tf.random.truncated_normal instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:192: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.\n",
      "\n",
      "WARNING:tensorflow:From C:\\Users\\jim19007\\Anaconda3\\envs\\visualization\\lib\\site-packages\\tensorflow_core\\python\\ops\\nn_impl.py:183: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:206: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:208: The name tf.train.GradientDescentOptimizer is deprecated. Please use tf.compat.v1.train.GradientDescentOptimizer instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:210: calling reduce_sum_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "keep_dims is deprecated, use keepdims instead\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:216: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:217: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:218: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:221: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:222: The name tf.summary.FileWriter is deprecated. Please use tf.compat.v1.summary.FileWriter instead.\n",
      "\n",
      "Initialized\n",
      "WARNING:tensorflow:From lib\\Building_Literature_Embedding_Model.py:259: The name tf.RunMetadata is deprecated. Please use tf.compat.v1.RunMetadata instead.\n",
      "\n",
      "Average loss at step  0 / 10 :  17.318113337367468\n",
      "time:  418.16795587539673\n",
      "Nearest to consequently: microm, fewer, syk, thymus, families, achieved, ally, delay,\n",
      "Nearest to lncrnas: ribosome, correction, factor, improves, gonadal, microscopy, anemia, kir,\n",
      "Nearest to demonstrates: understand, acidic, cd8, domains, p53, demonstrate, use, thaliana,\n",
      "Nearest to selection: methods, treated, #TBK1, lymphocyte, intron, #BACE1, clarify, pedigree,\n",
      "Nearest to absolute: family, oocytes, associated, presents, dimerization, cases, erk, facilitate,\n",
      "Nearest to rnas: bioinformatic, make, complicated, matrix, conformation, targeted, intestinal, genes,\n",
      "Nearest to gene: kir, dna, combined, strategy, markers, meiotic, improvement, genetic,\n",
      "Nearest to cerna: alt, membrane, degs, activated, recruits, cad, stratified, expressed,\n",
      "Nearest to receiver: directly, intractable, parallel, recombination, spectrum, gastric, selectively, selection,\n",
      "Nearest to survival: color, agent, metabolic, regulating, neovascularization, developing, solid, protein,\n",
      "Nearest to lncrna: demethylase, molecular, apolipoprotein, passive, mfs, high, materials, caused,\n",
      "Nearest to non-coding: term, hypoplasia, paclitaxel, significant, signaling, study, susceptibility, computational,\n",
      "Nearest to atlas: disease, transplant, mutations, rat, observed, impairments, embryo, oligomeric,\n",
      "Nearest to verified: kir, current, interpreted, overgrowth, tool, resistance, contractile, adp,\n",
      "Nearest to stage: left, control, sustained, distant, evident, counseling, apo, dfs,\n",
      "Nearest to score: sample, promote, tendon, conjugates, ca2, crpc, previously, bcor,\n",
      "Average loss at step  1 / 10 :  0.9283847747664211\n",
      "time:  386.4880259037018\n",
      "Nearest to consequently: microm, syk, fewer, mouth, families, poly, homolog, achieved,\n",
      "Nearest to lncrnas: correction, improves, anemia, factor, fdr, composite, #LAMA2, similarly,\n",
      "Nearest to demonstrates: acidic, extract, domains, understand, acts, use, cd8, ctbp1,\n",
      "Nearest to selection: #TBK1, treated, river, rural, intron, colocalized, methods, hcn,\n",
      "Nearest to absolute: oocytes, derivative, unit, glycoproteins, family, presents, dimerization, 1p36,\n",
      "Nearest to rnas: make, bioinformatic, #IL10, complicated, brown, conformation, protects, taxol,\n",
      "Nearest to gene: kir, combined, cleaves, genes, synthetase, #H4, #GAS5, derivatives,\n",
      "Nearest to cerna: alt, recruits, membrane, #PRSS56, activated, degs, #MKI67, nov,\n",
      "Nearest to receiver: directly, intractable, bhlh, evolved, parallel, selectively, #E2, spectrum,\n",
      "Nearest to survival: color, agent, neovascularization, regulating, protects, solid, developing, attenuation,\n",
      "Nearest to lncrna: apolipoprotein, demethylase, mfs, passive, o-acyltransferase, molecular, caused, #TLR7,\n",
      "Nearest to non-coding: ncoa4-ret, term, hypoplasia, chemistry, paclitaxel, #UGT2B7, upper, modified,\n",
      "Nearest to atlas: #LINC00467, transplant, rat, disease, siae, sea, #SAMHD1, transporters,\n",
      "Nearest to verified: interpreted, allele, mas, adp, bdnf, #LRRK2, kir, express,\n",
      "Nearest to stage: left, control, sustained, apo, distant, evident, wildtype, harbored,\n",
      "Nearest to score: conjugates, promote, tendon, sample, #H2, semaphorin, #PRDX1, ca2,\n",
      "Average loss at step  2 / 10 :  0.5995025641559834\n",
      "time:  407.6332976818085\n",
      "Nearest to consequently: microm, mouth, syk, venom, fewer, poly, thymus, nanotubes,\n",
      "Nearest to lncrnas: #LAMA2, correction, anemia, improves, ribosomes, mll, #MAPK1, microscopy,\n",
      "Nearest to demonstrates: acidic, understand, ctbp1, domains, #NUF2, extract, validates, use,\n",
      "Nearest to selection: #TBK1, rural, river, compendium, hcn, treated, intron, colocalized,\n",
      "Nearest to absolute: #HSPB1, derivative, oxt, glycoproteins, oocytes, cpa, 1p36, hlms,\n",
      "Nearest to rnas: make, #IL10, bioinformatic, complicated, taxol, conformation, #RPE65, run,\n",
      "Nearest to gene: genes, cleaves, kir, #H4, derivatives, #GAS5, tick, reln,\n",
      "Nearest to cerna: alt, #PRSS56, recruits, #MKI67, membrane, nov, degs, activated,\n",
      "Nearest to receiver: directly, intractable, bhlh, #FBXO33, p40, #-one, protein-2, evolved,\n",
      "Nearest to survival: color, agent, neovascularization, solvent, protects, regulating, attenuation, nov,\n",
      "Nearest to lncrna: demethylase, apolipoprotein, mfs, passive, o-acyltransferase, stratum, molecular, osteotomy,\n",
      "Nearest to non-coding: ncoa4-ret, chemistry, hypoplasia, term, #UGT2B7, paclitaxel, tethered, high-voltage,\n",
      "Nearest to atlas: #LINC00467, siae, transplant, shields, rat, #LINC00472, #SAMHD1, disease,\n",
      "Nearest to verified: tryptamine, interpreted, prodrug, mas, synthases, adp, hops, #LRRK2,\n",
      "Nearest to stage: left, apo, fbxo15, dels, control, wildtype, sustained, delivered,\n",
      "Nearest to score: conjugates, tendon, semaphorin, #PRDX1, #H2, promote, sample, sele,\n",
      "Average loss at step  3 / 10 :  0.5314434624305663\n",
      "time:  408.7071511745453\n",
      "Nearest to consequently: mouth, syk, microm, venom, fewer, nanotubes, thymus, #VPS35,\n",
      "Nearest to lncrnas: #LAMA2, #LMBR1, anemia, #GPX4, #397, improves, microscopy, correction,\n",
      "Nearest to demonstrates: acidic, ctbp1, #NUF2, understand, validates, domains, rab32, cos,\n",
      "Nearest to selection: #TBK1, rural, river, compendium, hcn, treated, methods, intron,\n",
      "Nearest to absolute: #HSPB1, oxt, hlms, cpa, derivative, glycoproteins, oocytes, 1p36,\n",
      "Nearest to rnas: make, #IL10, taxol, complicated, bioinformatic, patronin, #RPE65, targeted,\n",
      "Nearest to gene: genes, cleaves, tick, #H4, dmem, #SAMHD1, reln, cd11b,\n",
      "Nearest to cerna: #PRSS56, alt, #MKI67, recruits, nov, degs, #ODF1, sialylation,\n",
      "Nearest to receiver: intractable, directly, #FBXO33, bhlh, rab26, p40, #-one, protein-2,\n",
      "Nearest to survival: color, agent, nov, inward, attenuation, solvent, protects, #IL18,\n",
      "Nearest to lncrna: demethylase, mfs, o-acyltransferase, apolipoprotein, passive, osteotomy, stratum, lid,\n",
      "Nearest to non-coding: ncoa4-ret, chemistry, #UGT2B7, paclitaxel, hypoplasia, high-voltage, term, tethered,\n",
      "Nearest to atlas: #LINC00467, siae, transplant, shields, rat, #LINC00472, sea, haloperidol,\n",
      "Nearest to verified: tryptamine, interpreted, prodrug, mas, synthases, hops, segregating, adp,\n",
      "Nearest to stage: left, fbxo15, apo, dels, pneumococcal, harbored, wildtype, thresholds,\n",
      "Nearest to score: conjugates, semaphorin, tendon, #PRDX1, sample, #H2, sele, promote,\n",
      "Average loss at step  4 / 10 :  0.4924650083252683\n",
      "time:  405.08115553855896\n",
      "Nearest to consequently: mouth, syk, microm, venom, nanotubes, fewer, #HOOK1, sixth,\n",
      "Nearest to lncrnas: #LMBR1, #397, mainstay, #GPX4, #LAMA2, discs, #224, anemia,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, understand, validates, rab32, domains, gjb5,\n",
      "Nearest to selection: rural, river, #TBK1, compendium, areata, hcn, enschit00000007977-pomp, galphai,\n",
      "Nearest to absolute: #HSPB1, hlms, cpa, oxt, prg4, beijing-you, derivative, iso,\n",
      "Nearest to rnas: make, patronin, taxol, #IL10, complicated, bioinformatic, ats, acetic,\n",
      "Nearest to gene: genes, dmem, cleaves, tick, reln, identified, cd11b, #H4,\n",
      "Nearest to cerna: #PRSS56, alt, #MKI67, degs, #SSC7, nov, #ODF1, sialylation,\n",
      "Nearest to receiver: intractable, directly, #FBXO33, bhlh, rab26, #-one, p40, protein-2,\n",
      "Nearest to survival: color, agent, nov, 3-c, inward, delta, #GPX4, tfpi,\n",
      "Nearest to lncrna: demethylase, o-acyltransferase, progression, mfs, osteotomy, passive, lid, cancer,\n",
      "Nearest to non-coding: ncoa4-ret, chemistry, #UGT2B7, high-voltage, paclitaxel, #CTRB1, hypoplasia, sensu,\n",
      "Nearest to atlas: #LINC00467, siae, shields, haloperidol, #LINC00472, transplant, validations, nomenclature,\n",
      "Nearest to verified: tryptamine, interpreted, prodrug, mas, synthases, hops, diethylene, king,\n",
      "Nearest to stage: left, fbxo15, apo, dels, pneumococcal, thresholds, harbored, #PRMT6,\n",
      "Nearest to score: conjugates, semaphorin, tendon, sample, #PRDX1, sele, eif5a, #H2,\n",
      "Average loss at step  5 / 10 :  0.4678071203831536\n",
      "time:  405.21085476875305\n",
      "Nearest to consequently: mouth, syk, venom, microm, nanotubes, hanoi, #HOOK1, sixth,\n",
      "Nearest to lncrnas: mainstay, #397, #LMBR1, lncrna, #GPX4, lov, #LAMA2, 4-1bb,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, understand, validates, rab32, gjb5, domains,\n",
      "Nearest to selection: river, rural, #TBK1, compendium, areata, hcn, galphai, enschit00000007977-pomp,\n",
      "Nearest to absolute: hlms, #HSPB1, cpa, oxt, beijing-you, prg4, phosphocholine, iso,\n",
      "Nearest to rnas: make, patronin, taxol, complicated, #IL10, bioinformatic, ats, acetic,\n",
      "Nearest to gene: genes, dmem, reln, tick, cd11b, #H4, identified, insr,\n",
      "Nearest to cerna: #PRSS56, degs, alt, #MKI67, sialylation, #ODF1, nov, #SSC7,\n",
      "Nearest to receiver: intractable, directly, #FBXO33, rab26, bhlh, #-one, p40, reverted,\n",
      "Nearest to survival: agent, color, 3-c, nov, inward, #GPX4, tfpi, silver,\n",
      "Nearest to lncrna: progression, demethylase, cancer, o-acyltransferase, osteotomy, cp110, mfs, lid,\n",
      "Nearest to non-coding: ncoa4-ret, #UGT2B7, high-voltage, chemistry, paclitaxel, rs10444412-t, nanoclusters, #CTRB1,\n",
      "Nearest to atlas: #LINC00467, siae, haloperidol, shields, validations, #LINC00472, nomenclature, transplant,\n",
      "Nearest to verified: tryptamine, interpreted, mas, prodrug, diethylene, king, synthases, hops,\n",
      "Nearest to stage: left, fbxo15, apo, pneumococcal, dels, distant, #PRMT6, pact,\n",
      "Nearest to score: semaphorin, conjugates, sample, tendon, eif5a, #PRDX1, sele, #H2,\n",
      "Average loss at step  6 / 10 :  0.45390795381444016\n",
      "time:  397.94328141212463\n",
      "Nearest to consequently: mouth, syk, venom, hanoi, nanotubes, microm, #HOOK1, sixth,\n",
      "Nearest to lncrnas: lncrna, mainstay, #397, #LMBR1, #GPX4, noncoding, pioneered, 4-1bb,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, understand, validates, rab32, gjb5, domains,\n",
      "Nearest to selection: river, rural, areata, #TBK1, hcn, compendium, galphai, iia,\n",
      "Nearest to absolute: hlms, cpa, #HSPB1, oxt, beijing-you, prg4, pachon, phosphocholine,\n",
      "Nearest to rnas: make, patronin, promotes, electric, ats, acetic, taxol, complicated,\n",
      "Nearest to gene: genes, dmem, reln, identified, cd11b, mafk, analysis, #H4,\n",
      "Nearest to cerna: #PRSS56, degs, #MKI67, gon4l, alt, #ODF1, #SSC7, sialylation,\n",
      "Nearest to receiver: intractable, directly, #FBXO33, rab26, #-one, bhlh, p40, #BTNL3,\n",
      "Nearest to survival: agent, 3-c, inward, color, nov, #GPX4, database, prognostic,\n",
      "Nearest to lncrna: progression, lncrnas, cp110, proliferation, demethylase, tissues, osteotomy, o-acyltransferase,\n",
      "Nearest to non-coding: ncoa4-ret, rs10444412-t, #UGT2B7, nanoclusters, paclitaxel, high-voltage, chemistry, #CDH7,\n",
      "Nearest to atlas: #LINC00467, siae, validations, haloperidol, shields, #LINC00472, angular, #BTNL3,\n",
      "Nearest to verified: tryptamine, interpreted, mas, prodrug, diethylene, king, thr205, synthases,\n",
      "Nearest to stage: left, fbxo15, pneumococcal, dels, apo, pact, distant, #BIRC5,\n",
      "Nearest to score: sample, semaphorin, eif5a, conjugates, tendon, non-progressive, sele, eml4-alk,\n",
      "Average loss at step  7 / 10 :  0.44378286524477284\n",
      "time:  406.9059808254242\n",
      "Nearest to consequently: mouth, syk, hanoi, venom, microm, nanotubes, #HOOK1, ptgdr,\n",
      "Nearest to lncrnas: lncrna, #397, mainstay, #LMBR1, noncoding, #GPX4, pioneered, rnas,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, validates, understand, rab32, gjb5, #PSMC4,\n",
      "Nearest to selection: areata, river, rural, #TBK1, hcn, compendium, iia, galphai,\n",
      "Nearest to absolute: hlms, cpa, beijing-you, oxt, #HSPB1, pachon, prg4, phosphocholine,\n",
      "Nearest to rnas: make, patronin, promotes, electric, ats, acetic, #CCN2, complicated,\n",
      "Nearest to gene: genes, dmem, analysis, reln, tick, cd11b, mafk, sry,\n",
      "Nearest to cerna: #PRSS56, gon4l, degs, #MKI67, alt, #ODF1, sialylation, abcd3,\n",
      "Nearest to receiver: directly, intractable, #FBXO33, rab26, #-one, bhlh, p40, #BTNL3,\n",
      "Nearest to survival: agent, 3-c, prognostic, database, inward, nov, color, #GPX4,\n",
      "Nearest to lncrna: progression, lncrnas, proliferation, tissues, cp110, long, overexpression, demethylase,\n",
      "Nearest to non-coding: ncoa4-ret, rs10444412-t, nanoclusters, #UGT2B7, #CDH7, paclitaxel, high-voltage, #EBNA2,\n",
      "Nearest to atlas: siae, #LINC00467, validations, haloperidol, shields, #BTNL3, nomenclature, angular,\n",
      "Nearest to verified: tryptamine, interpreted, prodrug, diethylene, mas, thr205, king, jiang,\n",
      "Nearest to stage: fbxo15, left, pneumococcal, apo, dels, distant, pact, h-nmr,\n",
      "Nearest to score: sample, semaphorin, eif5a, non-progressive, asymmetry, #CUC2, eml4-alk, assassin,\n",
      "Average loss at step  8 / 10 :  0.42781765453789944\n",
      "time:  440.724595785141\n",
      "Nearest to consequently: mouth, hanoi, syk, venom, nanotubes, #HOOK1, microm, ptgdr,\n",
      "Nearest to lncrnas: lncrna, #397, mainstay, noncoding, #LMBR1, #GPX4, rnas, pioneered,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, validates, understand, rab32, gjb5, #HBA1,\n",
      "Nearest to selection: areata, rural, river, #TBK1, hcn, iia, a-to-i, compendium,\n",
      "Nearest to absolute: hlms, cpa, beijing-you, pachon, oxt, #HSPB1, prg4, phosphocholine,\n",
      "Nearest to rnas: make, patronin, promotes, lncrna, acetic, electric, rna, ats,\n",
      "Nearest to gene: genes, dmem, analysis, mafk, gpn, tick, cd11b, producer,\n",
      "Nearest to cerna: #PRSS56, gon4l, degs, abcd3, sialylation, #MKI67, #ODF1, turner,\n",
      "Nearest to receiver: directly, intractable, #FBXO33, rab26, #-one, bhlh, #BTNL3, p40,\n",
      "Nearest to survival: prognostic, agent, database, 3-c, inward, macrodontia, nov, color,\n",
      "Nearest to lncrna: lncrnas, progression, proliferation, long, cp110, overexpression, tissues, demethylase,\n",
      "Nearest to non-coding: ncoa4-ret, rs10444412-t, nanoclusters, #UGT2B7, paclitaxel, #CDH7, #EBNA2, high-voltage,\n",
      "Nearest to atlas: siae, validations, haloperidol, #LINC00467, shields, #BTNL3, nomenclature, #GRM4,\n",
      "Nearest to verified: tryptamine, diethylene, thr205, mas, prodrug, interpreted, #LYRM1, jiang,\n",
      "Nearest to stage: fbxo15, left, pneumococcal, apo, distant, pact, dels, h-nmr,\n",
      "Nearest to score: sample, semaphorin, eif5a, non-progressive, #CUC2, asymmetry, eml4-alk, assassin,\n",
      "Average loss at step  9 / 10 :  0.4186342307663809\n",
      "time:  404.7409768104553\n",
      "Nearest to consequently: mouth, hanoi, syk, venom, #HOOK1, nanotubes, ptgdr, khk,\n",
      "Nearest to lncrnas: lncrna, #397, noncoding, mainstay, #LMBR1, rnas, long, pioneered,\n",
      "Nearest to demonstrates: ctbp1, acidic, #NUF2, validates, understand, rab32, gjb5, thr373asnfster19,\n",
      "Nearest to selection: areata, rural, river, hcn, a-to-i, #TBK1, iia, galphai,\n",
      "Nearest to absolute: beijing-you, hlms, cpa, pachon, oxt, prg4, #HSPB1, phosphocholine,\n",
      "Nearest to rnas: make, rna, lncrna, patronin, promotes, lncrnas, overexpression, acetic,\n",
      "Nearest to gene: genes, mafk, analysis, tick, dmem, gpn, cd11b, parylation,\n",
      "Nearest to cerna: gon4l, #PRSS56, degs, abcd3, atp5g3, light-dependent, ultraconserved, sialylation,\n",
      "Nearest to receiver: directly, #FBXO33, intractable, rab26, #-one, bhlh, #BTNL3, #PKR1,\n",
      "Nearest to survival: prognostic, agent, database, 3-c, inward, macrodontia, nov, trng,\n",
      "Nearest to lncrna: lncrnas, progression, long, proliferation, overexpression, cp110, tissues, rnas,\n",
      "Nearest to non-coding: ncoa4-ret, rs10444412-t, paclitaxel, nanoclusters, #UGT2B7, #CDH7, #EBNA2, high-voltage,\n",
      "Nearest to atlas: validations, haloperidol, siae, #LINC00467, #BTNL3, shields, c3a, #GRM4,\n",
      "Nearest to verified: tryptamine, diethylene, thr205, #LYRM1, prodrug, mas, jiang, interpreted,\n",
      "Nearest to stage: fbxo15, h-nmr, left, pact, pneumococcal, apo, distant, #PSD95,\n",
      "Nearest to score: sample, #CUC2, non-progressive, semaphorin, assassin, eif5a, asymmetry, #ALS2,\n"
     ]
    }
   ],
   "source": [
    "#print(\"=== Creating Training Data For Fig. 2 ===\")\n",
    "eg.creating_training_data_for_gene2doc(context_size)\n",
    " \n",
    "#print(\"=== Starting Model Training For Fig. 2 ===\")\n",
    "eg.model_setting(dimension=dimension, num_sampled=num_sampled)\n",
    "eg.starting_sorting(model_path)\n",
    "eg.model_training(epoch=epoch, batch_size=batch_size, window_size = context_size)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "271d8e4d-d850-4034-a537-c417eaca298e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
